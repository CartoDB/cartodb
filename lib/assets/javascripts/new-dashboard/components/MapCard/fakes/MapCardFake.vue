<template>
  <div>
    <CondensedMapCardFake v-if="condensed"></CondensedMapCardFake>
    <SimpleMapCardFake v-else></SimpleMapCardFake>
  </div>
</template>

<script>
import SimpleMapCardFake from 'new-dashboard/components/MapCard/fakes/SimpleMapCardFake';
import CondensedMapCardFake from 'new-dashboard/components/MapCard/fakes/CondensedMapCardFake';

export default {
  name: 'MapCardFake',
  components: {
    SimpleMapCardFake,
    CondensedMapCardFake
  },
  props: {
    condensed: {
      type: Boolean,
      default: false
    }
  }
};
</script>
