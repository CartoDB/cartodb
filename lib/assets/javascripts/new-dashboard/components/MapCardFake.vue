<template>
  <div class="map-fake-card">
    <div class="image-placeholder"></div>
    <div class="text-container-placeholder">
      <div class="text-placeholder"></div>
      <div class="text-placeholder text-placeholder--double"></div>
      <div class="text-placeholder"></div>
      <div class="text-placeholder"></div>
      <div class="text-placeholder"></div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'MapCardFake'
};
</script>

<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.map-fake-card {
  border: 1px solid $light-grey;
  border-radius: 2px;
  background-color: $white;
}

.image-placeholder {
  width: 100%;
  height: 140px;
  background-color: $softblue;
}

.text-container-placeholder {
  padding: 24px 16px;
}

.text-placeholder {
  width: 100%;
  height: 24px;
  margin-bottom: 8px;
  background-color: $softblue;

  &.text-placeholder--double {
    height: 48px;
  }

  &:last-of-type {
    margin-bottom: 0;
  }
}
</style>
