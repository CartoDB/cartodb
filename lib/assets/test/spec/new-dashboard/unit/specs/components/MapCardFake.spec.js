import { shallowMount } from '@vue/test-utils';
import MapCardFake from 'new-dashboard/components/MapCardFake';

describe('MapCardFake.vue', () => {
  it('should render correct contents', () => {
    const mapCardFake = shallowMount(MapCardFake);
    expect(mapCardFake).toMatchSnapshot();
  });
});
