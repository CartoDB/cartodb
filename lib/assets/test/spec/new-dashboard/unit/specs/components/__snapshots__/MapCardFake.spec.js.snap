// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapCardFake.vue should render correct contents 1`] = `
<div class="map-fake-card">
  <div class="image-placeholder"></div>
  <div class="text-container-placeholder">
    <div class="text-placeholder"></div>
    <div class="text-placeholder text-placeholder--double"></div>
    <div class="text-placeholder"></div>
    <div class="text-placeholder"></div>
    <div class="text-placeholder"></div>
  </div>
</div>
`;
